File: Programmers/Inho/C#/Level01/하샤드 수.cs
using System;
public class Solution {
    public bool solution(int x) {
        bool answer = true;
        int na = x%10;
        int moc = x/10;
        int ha = na;
        int now = x/10;
        while(moc!=0){
            na = now%10;
            moc = now/10;
            now = now/10;
            ha = ha+na;
        }
        ha += moc;
        if(x%ha==0){
            answer = true;
        } else
            answer = false;
        return answer;
    }
}

File: Programmers/Inho/C++/Level01/하샤드 수.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

bool solution(int x) {
    bool answer = true;
    int hashad = 0;
    int before = x;
    while(true){
        hashad += x%10;
        cout << "hashad: " << hashad << endl;
        if(x/10==0)
            break;
        x = x/10;
    }
    if(before%hashad==0)
        answer = true;
    else
        answer = false;

    return answer;
}